### ISSUE TITLE
[Schedulers] Add methods for conversion between noise and denoised predictions

### ISSUE THREAD

**Issue by hlky** [huggingface/diffusers#9629]
Guidance methods like APG (#9585 #9626) require conversion between noise and denoised predictions. The computation of these predictions varies across schedulers and depends on `prediction_type`. Currently, this is embedded in `step` functions of the schedulers which complicates implementation of APG.

Adding separate methods for conversion between noise and denoised predictions would simplify implementation and allow easier experimentation.

**Comment by yiyixuxu** [comment]
Do you need a function like this we have for dpm? https://github.com/huggingface/diffusers/blob/07bd2fabb60dbd1c7da6fd176c62e592402b19e3/src/diffusers/schedulers/scheduling_dpmsolver_multistep.py#L594

**Comment by hlky** [comment]
Yes like that, there's currently also something similar in UniPC, they'd need to be controllable with a parameter rather than config or internal variables, and handle different sigmas for 2nd order schedulers like Heun and KDPM2

**Comment by yiyixuxu** [comment]
Make sense! 

Do you want to open up a PR for UniPC? We'd need to keep this method consistent across all schedulers that have them, so we need to use the same name and signature and return the same output etc 

For the rest of the schedulers, we can also open this task up to communities if you don't have much time: we would need to run slow tests for each scheduler we refactor here, so it might be a bit time-consuming for one person to work on!

**Comment by hlky** [comment]
Sure, I'll start with UniPC. I don't mind doing the rest of the schedulers, I don't have much else going on, more community contributions in schedulers would be good though!
